src/code/z_map_exp.c

```diff
@@ -99,66 +112,75 @@ s32 Map_GetBossAreaIndex(PlayState* play) {
 }
 
 /**
- * Returns true if the current scene is a boss scene, false otherwise.
+ * Checks if the current scene is a boss scene.
+ * @return true if the current scene is a boss scene, false otherwise.
  */
-s32 Map_IsInBossArea(PlayState* play) {
-    if (Map_GetBossAreaIndex(play) == -1) {
+s32 Map_IsInBossScene(PlayState* play) {
+    if (MapExp_TryGetDungeonSceneIndexForBoss(play) == -1) {
         return false;
     }
 
     return true;
 }
 
-s32 D_801BF5A4[] = {
-    SCENE_22DEKUCITY,
-    SCENE_KOEPONARACE,
-    SCENE_F01,
-};
-
-s32 func_8010A238(PlayState* play) {
+/**
+ * Test if the current scene is in a specific list of scenes.
+ * This function may be incomplete.
+ * @return -1 if not in the list, else returns the DungeonSceneIndex for that scene
+ */
+s32 MapExp_TryGetDungeonSceneIndexForUnknownBehavior(PlayState* play) {
```

Comment:
```suggestion
s32 Map_GetDungeonSceneIndexOverworld(PlayState* play) {
```
Though it isn't all overworld scenes, would just add that only a few overworld areas have a `dungeonSceneIndex`